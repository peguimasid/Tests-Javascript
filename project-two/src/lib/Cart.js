export class Cart {
  constructor() {
    this.itens = [];
  }

  getTotal() {
    return this.itens.reduce((acc, { product, quantity }) => {
      return acc + product.price * quantity;
    }, 0);
  }

  add(item) {
    this.itens.push(item);
  }
}
