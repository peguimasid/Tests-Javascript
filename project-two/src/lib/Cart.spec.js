import { Cart } from "./Cart";

describe("Cart", () => {
  let cart = new Cart();

  const product = {
    title: "Airpods Pro",
    price: 35388,
  };

  const product2 = {
    title: "Airpods Max",
    price: 98745,
  };

  beforeEach(() => {
    cart = new Cart();
  });

  it("should return 0 when getTotal() is executed in a newly created instance", () => {
    expect(cart.getTotal()).toEqual(0);
  });

  it("should multipy quantity and price and return the total amount", () => {
    const item = {
      product: {
        title: "Airpods Pro",
        price: 35388,
      },
      quantity: 2,
    };

    cart.add(item);
    expect(cart.getTotal()).toEqual(70776);
  });

  it("should ensure that a product appear just once in cart", () => {
    cart.add({
      product,
      quantity: 2,
    });

    cart.add({
      product,
      quantity: 1,
    });

    expect(cart.getTotal()).toEqual(35388);
  });

  it("should update total when a product its includes and then removed from card", () => {
    cart.add({
      product,
      quantity: 2,
    });

    cart.add({
      product: product2,
      quantity: 2,
    });

    cart.remove(product);

    expect(cart.getTotal()).toEqual(197490);
  });
});
