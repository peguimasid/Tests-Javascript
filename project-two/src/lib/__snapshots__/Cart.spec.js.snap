// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart checkout() should return an object with the total and the list of items 1`] = `
{
  "items": [
    {
      "product": {
        "price": 35388,
        "title": "Airpods Pro",
      },
      "quantity": 2,
    },
    {
      "product": {
        "price": 98745,
        "title": "Airpods Max",
      },
      "quantity": 3,
    },
  ],
  "total": 367011,
}
`;

exports[`Cart summary() should return an object with the total and the list of items when is called 1`] = `
{
  "items": [
    {
      "product": {
        "price": 35388,
        "title": "Airpods Pro",
      },
      "quantity": 2,
    },
    {
      "product": {
        "price": 98745,
        "title": "Airpods Max",
      },
      "quantity": 3,
    },
  ],
  "total": 367011,
}
`;
